R/object_name_linter.R

```diff
@@ -10,7 +10,7 @@ object_name_xpath <- local({
   #   is not possible for strings, though we do still have to
   #   be aware of cases like 'a$"b" <- 1'.
   xp_assignment_target_fmt <- paste0(
-    "not(preceding-sibling::OP-DOLLAR)",
+    "not(parent::expr[OP-DOLLAR or OP-AT])",
```

Comment:
I would have expected this to change behavior on `badName$a <- 3`. But the tests have us covered 😃

Reply:
yep! different `<expr>`. I think the parser is being "smart" about not dropping the RHS of `$`/`@` into a child `<expr>` because only `STR_CONST`/`SYMBOL` are parseable. But this subverts our usual expectation about where to find `SYMBOL` nodes, hence clauses like this.